mantidproject/vesuvio
# Improve third derivative

**Description of work:**
Previous third derivative result was padded with 6 zeros on each side.
This improvement extends the range and the function by 6 indices on each side, so that when the third derivative is taken, the size of the resulting array is 12 indices less than the input (6 on each side).

So in essence this just means that I am extending the range before taking the derivative so that after the derivative the size matches the size before the extension.

This just makes sure the derivative is continuous and does not including zero padding, which contributes to a smoother ncp profile (orange line is corrected):

![third-derivative-smooth](https://github.com/user-attachments/assets/7150cc68-2dcd-41e9-9084-c06f2b28d89c)

**To test:**
Activate an environment with mantidworkbench installed:
  - Checkout this branch and install with `pip install .`
  - run `mvesuvio config`
  - Open Mantid workbench and open file `~/.mvesuvio/analysis_inputs.py`
  - Click Run
  - Plot workspace `analysis_inputs_fwd_0_1.0079_fse_sum` and check that curve is smooth at the edges
